### PR TITLE
Add network/config tests

### PR DESCRIPTION
**Pull request by sshumakov**
Added unit tests for network/config. #6

### CODE REVIEW

File: network/config_test.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    "go.uber.org/zap"
+    "io"
+)
+
+func TestSetAddress(t *testing.T) {
+    config := DefaultConfig
+    addr := "192.168.4.62"
+    
+    setFunc := SetAddress("192.168.4.62")
+    setFunc(&config)
```
Comment (sshumakov) [inline network/config_test.go]: No, there is no value doing it in two steps. Didn't know that you can do it in one.

File: network/config_test.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+    if config.minPeers != minPeers {
+        t.Fatalf("Expected to set minPeers to %d. Actual: %d", minPeers, config.minPeers)
+    }
+}
+
+func TestSetNetwork(t *testing.T) {
+    config := DefaultConfig
+    network := make([]byte, 2)
```
Comment (sshumakov) [inline network/config_test.go]: Yea, initializer improves readability a lot.

File: network/config_test.go [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    setFunc(&config)
+
+    if config.subscriber != subscriber {
+        t.Fatalf("Expected to set subscriber to %v. Actual: %v", subscriber, config.subscriber)
+    }
+}
+
+func TestSetTimeout(t *testing.T) {
+    config := DefaultConfig
+    timeout := time.Duration(5)
```
Comment (sshumakov) [inline network/config_test.go]: Yea, I was mostly focused on checking the end result, not values we specify. Going to change it however.

**Comment by sshumakov** [comment]
Pushed changes, please take look.

File: network/config_test.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+import (
+    "testing"
+    "time"
+    "go.uber.org/zap"
+    "io"
+    "github.com/stretchr/testify/assert"
+)
+
+func TestSetAddress(t *testing.T) {
+    config := DefaultConfig
```
Comment (awishformore) [inline network/config_test.go]: I still think it's dangerous to use the `DefaultConfig` here. By some coincidence, if the default value is the same as the one you set using the helper function, then you would end up with a defunct test case. If you simply use `config := Config{}`, then you are certain all values are default initialized and there is no such possibility of conflict.

Comment (sshumakov) [inline network/config_test.go]: Good point, pushing changes.

**Pull request merged**